### PR TITLE
[ArchiveFilesV2] Fixed 'The system cannot find the file specified' error when root folder is parent for archive target path 

### PR DESCRIPTION
**Pull request by anatolybolshakov**
**Task name**: ArchiveFilesV2

**Description**: there was an issue with ArchiveFilesV2 task when the 'rootFolderOrFile' was the parent directory for 'archiveFile' path, 'replaceExistingArchive' was set up and the target archive already existed - archive file was found during files scanning phase in root folder, but removed later and not found during compressing.

Changed the order of phases (scanning files/removing of already existing archive) - to consider this case.

**Documentation changes required:** N

**Added unit tests:** Y - checked that tests fail before PR changes.

**Attached related issue:** N/A

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected - tested locally on simple scenarios

### CODE REVIEW

File: Tasks/ArchiveFilesV2/archivefiles.ts [hunk 1]
```diff
@@ -350,6 +340,17 @@ function doWork() {
             }
         }
 
+        // Find matching archive files
+        var files: string[] = findFiles();
+        utils.reportArchivePlan(files).forEach(function (line) {
+            console.log(line);
+        });
```
Comment (alexander-smolyakov) [inline Tasks/ArchiveFilesV2/archivefiles.ts:347]:
```suggestion
        utils.reportArchivePlan(files).forEach(line => console.log(line));
```

File: Tasks/ArchiveFilesV2/archivefiles.ts [hunk 2]
```diff
@@ -350,6 +340,17 @@ function doWork() {
...
+
+        tl.debug('Listing all ' + files.length + ' files to archive:');
```
Comment (alexander-smolyakov) [inline Tasks/ArchiveFilesV2/archivefiles.ts:349]:
```suggestion
        tl.debug(`Listing all ${files.length} files to archive:`);
```

File: Tasks/ArchiveFilesV2/archivefiles.ts [hunk 3]
```diff
@@ -350,6 +340,17 @@ function doWork() {
...
+        for (var i = 0; i < files.length; i++) {
+            tl.debug(files[i]);
+        }
```
Comment (alexander-smolyakov) [inline Tasks/ArchiveFilesV2/archivefiles.ts:352]:
```suggestion
        files.forEach(file => tl.debug(file));
```

**Review by denis-tikhomirov — APPROVED** [review]

**Review by alexander-smolyakov — APPROVED** [review]
LGTM and please take a look at the suggestions below:

**Review by EzzhevNikita — APPROVED** [review]

**Pull request merged**